lib/media.js

```diff
@@ -272,6 +282,37 @@ function getScriptFiles(state) {
     .map(pathJoin(assetHost, assetPath, assetDir));
 }
 
+/**
+ * Read the asset manifest generated by the `clay pack` bundling command and
+ * check for scripts which match the rendered component names. The manifest for
+ * each asset lists its complete set of dependencies, so we dedupe them.
```

Comment:
Why are we deuping anything here? Shouldn't that be webpack's job?

Reply:
Yup. It normally would be. The problem is, Webpack expects to be in control of the entire lifecycle of its assets: every bundle starts with one entrypoint, and the runtime takes it from there. Taking some of that control away in order to integrate with Clay makes things a little harder on us: we can't depend on Webpack to know which components are present on the Clay page and which scripts it should load to support them.

The workaround, which I sketched very quickly in [this gist](https://gist.github.com/phyllisstein/8f73f8ceb142a1cb0c86038f48e7c765), is to tell Webpack that _any_ component is a potential entrypoint. This way, we get an exhaustive list of its dependencies. But because the bundle _really_ a graph, we have to unwind the list and remove duplicates to ensure we don't wind up parsing and executing Lodash over and over and over.

Our next critical milestone after shipping this should be to move Clay's CSS into this pipeline. But after that, we'd do well to spend a little more time reverse-engineering Webpack's internal module loading code, in hopes of making this a little less brittle.